cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/query/QueryTree.kt

```diff
@@ -415,3 +415,18 @@ fun List<QueryTree<Boolean>>.mergeWithAny(
         assumptions = assumptions,
     )
 }
+
+infix fun <T : Any> T.eq(other: T): QueryTree<Boolean> {
+    val thisQuery = this.toQueryTree()
+    val result = thisQuery.value == other
+    return QueryTree(
+        result,
+        mutableListOf(thisQuery, QueryTree(other)),
+        "${thisQuery.value} == $other",
+        thisQuery.node,
+    )
+}
+
+fun <T : Any> T.toQueryTree(): QueryTree<T> {
```

Comment:
If `T` is already a `QueryTree`, it shouldn't do anything I guess but that's hard/impossible to express with the generics